src/cli/completion.rs

```diff
@@ -2,39 +2,98 @@ use crate::cli::Run;
 use anyhow::Result;
 use clap::{Parser, CommandFactory};
 use clap_complete::{generate, shells::Bash, shells::Zsh, shells::Fish};
+use std::fmt;
 use std::io;
 
 #[derive(Parser, Debug)]
+#[clap(after_help = "INSTALLATION INSTRUCTIONS:
+
+Bash:
+  $ sage completion bash > ~/.bash_completion.d/sage
+  # Make sure the directory exists first:
+  # mkdir -p ~/.bash_completion.d
+  # Add to ~/.bashrc if not already sourcing completion directory:
+  # source ~/.bash_completion.d/sage
+
+Zsh:
+  $ mkdir -p ~/.zsh/completions
+  $ sage completion zsh > ~/.zsh/completions/_sage
+  # Add to ~/.zshrc if not already in fpath:
+  # fpath=(~/.zsh/completions $fpath)
+  # autoload -U compinit && compinit
+
+Fish:
+  $ sage completion fish > ~/.config/fish/completions/sage.fish")]
 pub struct CompletionArgs {
-    /// Shell to generate completions for (bash, zsh, fish)
-    #[clap(value_enum)]
+    /// Shell to generate completions for
+    #[clap(value_enum, long_help = "Specifies which shell's completion script to generate. Choose from:
+- bash: Generates Bash completions that work with bash-completion
+- zsh: Generates Zsh completions using the _describe completer
+- fish: Generates Fish shell completions
+
+The generated script will be output to stdout, which you can redirect to the appropriate location for your shell.")]
     pub shell: Shell,
 }
 
 #[derive(clap::ValueEnum, Clone, Debug)]
+#[clap(rename_all = "lowercase")]
 pub enum Shell {
+    /// Bash shell completions (compatible with bash-completion)
     Bash,
+    /// Zsh shell completions (using the _describe completer)
     Zsh,
+    /// Fish shell completions
     Fish,
 }
 
+impl fmt::Display for Shell {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Shell::Bash => write!(f, "Bash"),
+            Shell::Zsh => write!(f, "Zsh"),
+            Shell::Fish => write!(f, "Fish"),
+        }
+    }
+}
+
 impl Run for CompletionArgs {
     async fn run(&self) -> Result<()> {
         let mut cmd = crate::cli::Cmd::command();
         let mut stdout = io::stdout();
 
+        // Print a helpful comment at the top of the generated script
         match self.shell {
             Shell::Bash => {
+                eprintln!("# Bash completion script for sage");
+                eprintln!("# Save this output to ~/.bash_completion.d/sage");
+                eprintln!("# Make sure the directory exists: mkdir -p ~/.bash_completion.d/");
+                eprintln!("# Add to ~/.bashrc: source ~/.bash_completion.d/sage");
+                eprintln!("#");
```

Comment:
If we use `println!` instead of `eprintln!` then it will use the stdout, rather than the stderr.
I think that might be better